### PR TITLE
Add Pico2 Tutorials on Raspberry Pi

### PR DESCRIPTION
**Pull request by psiddh**
### Summary
Add Pico2 Tutorials on Raspberry Pi 

[PLEASE REMOVE] See [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests) for ExecuTorch PR guidelines.

[PLEASE REMOVE] If this PR closes an issue, please add a `Fixes #<issue-id>` line.

[PLEASE REMOVE] If this PR introduces a fix or feature that should be the upcoming release notes, please add a "Release notes: <area>" label. For a list of available release notes labels, check out [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests).

### Test plan
[PLEASE REMOVE] How did you test this PR? Please write down any manual commands you used and note down tests that you have written if applicable.

### CODE REVIEW

File: docs/source/pico2_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# Pico2: A simple MNIST Tutorial
+
+Deploy your PyTorch models directly to Raspberry Pi Pico2 microcontroller with ExecuTorch and Infer.
```
Comment (GregoryComer) [inline docs/source/pico2_tutorial.md]: Is Infer just referring to inference in this case, or is it some specific noun? Wording seems to imply that Infer is a separate tool or library.

File: docs/source/pico2_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+## Step 1: Generate pte from given example Model
+
+- Use the [provided example model](https://github.com/pytorch/executorch/blob/main/examples/raspberry_pi/pico2/export_mlp_mnist.py)
+
+```bash
+python export_mlp_mnist.py # Creates balanced_tiny_mlp_mnist.pte
+```
+
+- **Note:** This is hand-crafted MNIST Classifier (proof-of-concept), and not production trained.This tiny MLP recognizes digits 0, 1, 4, and 7 using manually designed feature detectors.
```
Comment (GregoryComer) [inline docs/source/pico2_tutorial.md]: Nit: Missing a space in "production trained.This..."?

File: docs/source/pico2_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+arm-none-eabi-readelf -l examples/raspberry_pi/pico2/build/executorch_pico.elf
+```
+
+## Next Steps
+
+### Scale up your deployment
+
+- Train real models → Replace hand-crafted weights
+- Optimize further → INT8 quantization, pruning
+- Deploy fleet → Mass production ready
```
Comment (GregoryComer) [inline docs/source/pico2_tutorial.md]: Nit: Reads slightly weird.

File: examples/raspberry_pi/pico2/README.md [hunk 4]
```diff
@@ -4,44 +4,48 @@ This document outlines the steps required to run a simple MNIST digit recognitio
...
 
-Export using `torch.export()` and `to_edge()`
-Build firmware with one command
-Deploy directly to Pico2
+### Important Caveats
 
-#### Important Caveats:
-
-- Memory constraints - Models must fit in 520KB SRAM
+- Memory constraints - Models must fit in 520KB SRAMn (Pico2)
```
Comment (GregoryComer) [inline examples/raspberry_pi/pico2/README.md]: Is the n intended in SRAMn?

**Review by GregoryComer — APPROVED** [review]
Looks good to me. I left a few small nits comments.

File: examples/raspberry_pi/pico2/README.md (hunk 4 above)
Comment (psiddh) [reply examples/raspberry_pi/pico2/README.md]: fixed

File: docs/source/pico2_tutorial.md (hunk 3 above)
Comment (psiddh) [reply docs/source/pico2_tutorial.md]: Worded it better

File: docs/source/pico2_tutorial.md (hunk 1 above)
Comment (psiddh) [reply docs/source/pico2_tutorial.md]: Removed Infer
